hydra-node/test/Hydra/ModelSpec.hs

```diff
@@ -145,6 +148,16 @@ prop_generateTraces actions =
           hydraState st /= Start
             & counterexample ("state: " <> show st)
 
+prop_doesNotGenerate0AdaUTxO :: Actions WorldState -> Bool
+prop_doesNotGenerate0AdaUTxO (Actions actions) =
+  not (any contains0AdaUTxO actions)
+ where
+  contains0AdaUTxO :: Step WorldState -> Bool
+  contains0AdaUTxO = \case
+    _anyVar := Model.Commit _anyParty utxos -> any contains0Ada utxos
+    _anyOtherStep -> False
```

Comment:
Technically, I think we might want also to ensure there's no 0 value `NewTx` being generated but we might leave that aside for the moment.

Reply:
Good point. I'm wondering what it would mean, in real life, to send a tx for a 0 Ada payment. I guess it would be rejected by the current version of cardano-node.

See commit 826a417 for a proposal to include that in the code.

I'm just wondering then, does the name `prop_doesNotGenerate0AdaUTxO` still make sense?

Creating a new transaction with a payment of 0 Ada in the model would be equivalent to generating a 0 Ada UTxO right? If so, the name is good as is. If not, I would need to clarify the code and maybe event have a different, dedicated property test just for that.

What do you think @abailly-iohk ?

Reply:
I think that the way the code is currently written, you cannot generate `NewTx` with 0 value because each `NewTx` `Payment` consumes a UTxO in full thus if we always generate non-0 UTxO, we generate non-0 `NewTx`. Hence my suggestion to leave that aside: It will become a problem if we start consuming _parts_ of a UTxO.